File: packages/oidc/android/src/main/java/com/pingidentity/rnoidc/OidcErrorCodes.kt
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.rnoidc

/**
 * Stable error codes emitted by the OIDC module.
 *
 * @remarks
 * Keep these in sync with JS `OidcErrorCode`.
 */
object OidcErrorCodes {
  const val OIDC_AUTHORIZE_ERROR = "OIDC_AUTHORIZE_ERROR"
  const val OIDC_HAS_USER_ERROR = "OIDC_HAS_USER_ERROR"
  const val OIDC_TOKEN_ERROR = "OIDC_TOKEN_ERROR"
  const val OIDC_REFRESH_ERROR = "OIDC_REFRESH_ERROR"
  const val OIDC_USERINFO_ERROR = "OIDC_USERINFO_ERROR"
  const val OIDC_REVOKE_ERROR = "OIDC_REVOKE_ERROR"
  const val OIDC_LOGOUT_ERROR = "OIDC_LOGOUT_ERROR"
}

File: packages/oidc/ios/OidcErrorCodes.swift
//
//  OidcErrorCodes.swift
//  RNPingOidc
//
//  Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
//
//  This software may be modified and distributed under the terms
//  of the MIT license. See the LICENSE file for details.
//

import Foundation

/// Stable error codes emitted by the OIDC module.
///
/// Keep these in sync with JS `OidcErrorCode` and Android `OidcErrorCodes`.
enum OidcErrorCodes: String {
  case authorizeError = "OIDC_AUTHORIZE_ERROR"
  case hasUserError = "OIDC_HAS_USER_ERROR"
  case stateError = "OIDC_STATE_ERROR"
  case tokenError = "OIDC_TOKEN_ERROR"
  case refreshError = "OIDC_REFRESH_ERROR"
  case userinfoError = "OIDC_USERINFO_ERROR"
  case revokeError = "OIDC_REVOKE_ERROR"
  case logoutError = "OIDC_LOGOUT_ERROR"
}
